utils/tag_ptr/src/lib.rs

```diff
@@ -56,7 +64,8 @@ impl<T> Tagged<T> {
     /// # Safety
     ///
     /// - `T` must be non null.
-    pub(crate) const unsafe fn from_ptr(ptr: *mut T) -> Self {
+    #[inline]
+    pub const unsafe fn from_ptr(ptr: *mut T) -> Self {
         debug_assert!(align_of::<T>() >= 2);
```

Comment:
You should also replace the rest of the debug_assert with the compile time check.